OrderService/OrderService.Application/UseCases/CartUseCases/AddProductToCartUseCase/AddProductToCartRequestHandler.cs

```diff
@@ -9,18 +9,25 @@ internal class AddProductToCartRequestHandler(ITemporaryStorageService temporary
 {
     public async Task Handle(AddProductToCartRequest request, CancellationToken cancellationToken)
     {
-        var product = await productService.GetByIdIfSufficientQuantityAsync(request.product.Id, request.product.Quantity);
+        var cartProductQuantity = 0;
+
+        var cart = await temporaryStorage.GetCartAsync(cancellationToken);
+
+        if (cart.ContainsKey(request.product.Id))
+        {
+            cartProductQuantity = cart[request.product.Id].Quantity;
+        }
```

Comment:
what do you think about moving this to a separate method which returns result ProductQuantity?